Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/InMemorySpawnableAssetUtils.cpp

```diff
@@ -0,0 +1,128 @@
+
+        m_prefabSystemComponentInterface = AZ::Interface<PrefabSystemComponentInterface>::Get();
+        AZ_Assert(m_prefabSystemComponentInterface, "InMemorySpawnableAssetProcessor - Could not retrieve instance of PrefabSystemComponentInterface");
+
+        m_loaderInterface = AZ::Interface<Prefab::PrefabLoaderInterface>::Get();
+        AZ_Assert(m_loaderInterface, "InMemorySpawnableAssetProcessor - Could not retrieve instance of PrefabLoaderInterface");
+
+        m_stackProfile = stackProfile;
+        return m_converter.LoadStackProfile(m_stackProfile);
+    }
+
+    void InMemorySpawnableAssetProcessor::Deactivate()
+    {
+        m_assetContainer.ClearAllInMemorySpawnableAssets();
+        m_stackProfile = "";
+        m_loaderInterface = nullptr;
+        m_prefabSystemComponentInterface = nullptr;
+    }
+
+    bool InMemorySpawnableAssetProcessor::IsActivated() const
+    {
+        return m_converter.IsLoaded();
+    }
+
+    AZStd::string_view InMemorySpawnableAssetProcessor::GetStackProfile() const
+    {
+        return m_stackProfile;
+    }
+
+    AzFramework::InMemorySpawnableAssetContainer::CreateSpawnableResult InMemorySpawnableAssetProcessor::CreateInMemorySpawnableAsset(
+        AzToolsFramework::Prefab::TemplateId templateId, AZStd::string_view spawnableName, bool loadReferencedAssets)
+    {
+        if (!IsActivated())
+        {
+            return AZ::Failure(AZStd::string::format("Failed to create a prefab processing stack from key '%.*s'.", AZ_STRING_ARG(m_stackProfile)));
+        }
+
+        if (m_assetContainer.HasInMemorySpawnableAsset(spawnableName))
+        {
+            return AZ::Failure(AZStd::string::format("In-memory Spawnable '%.*s' already exists.", AZ_STRING_ARG(spawnableName)));
+        }
+
+        TemplateReference templateReference = m_prefabSystemComponentInterface->FindTemplate(templateId);
+        if (!templateReference.has_value())
+        {
+            return AZ::Failure(AZStd::string::format("Could not get Template DOM for given Template's id %llu .", templateId));
+        }
+
+        // Use a random uuid as this is only a temporary source.
+        PrefabConversionUtils::PrefabProcessorContext context(AZ::Uuid::CreateRandom());
+        PrefabDocument document(spawnableName);
+        document.SetPrefabDom(templateReference->get().GetPrefabDom());
+        context.AddPrefab(AZStd::move(document));
+        m_converter.ProcessPrefab(context);
+
+        if (!context.HasCompletedSuccessfully() || context.GetProcessedObjects().empty())
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Failed to convert the prefab into assets. Please confirm that the '%.*s' prefab processor stack is capable of producing a usable product asset.",
+                AZ_STRING_ARG(PrefabConversionUtils::IntegrationTests)));
```

Comment:
Is `context.GetProcessedObjects().empty()` always a failure? Is there a case where empty is valid?

Reply:
@michabr would know more, but I think this method turns a prefab, (the editor-time asset), and into its the products (spawnable and possibly network.spawnable). In every case at least a spawnable, so its products (aka processedobjects) should be at least size = 1